pkg/ccl/backupccl/backup_test.go

```diff
@@ -3869,6 +3871,7 @@ func TestProtectedTimestampsDuringBackup(t *testing.T) {
 
 	// We should have refused to GC over the timestamp which we needed to protect.
 	gcTable(true /* skipShouldQueue */)
+	runner.Exec(t, "SELECT * FROM foo AS OF SYSTEM TIME "+tsBefore)
```

Comment:
huh, is this require to get it to fail if the backup doesn't protect its TS? I'd have expected the ExportRequest to hit it when we close allowResponse below, right? Oh, I wonder if it is because we're using a *Response* filter, not a Request filter, so it has eagerly executed before GC?

Reply:
I ~think that it is required to fail with `kv.gc_ttl.strict_enforcement.enabled` which is set by default. I would prefer to test this more explicitly with an ExportRequest hitting the GC threshold. Will try updating it with a request filter.